genesyscloud/integration_facebook/resource_genesyscloud_integration_facebook_schema.go

```diff
@@ -0,0 +1,124 @@
+*/
+const resourceName = "genesyscloud_integration_facebook"
+
+// SetRegistrar registers all of the resources, datasources and exporters in the package
+func SetRegistrar(regInstance registrar.Registrar) {
+	regInstance.RegisterResource(resourceName, ResourceIntegrationFacebook())
+	regInstance.RegisterDataSource(resourceName, DataSourceIntegrationFacebook())
+	regInstance.RegisterExporter(resourceName, IntegrationFacebookExporter())
+}
+
+// ResourceIntegrationFacebook registers the genesyscloud_integration_facebook resource with Terraform
+func ResourceIntegrationFacebook() *schema.Resource {
+
+	return &schema.Resource{
+		Description: `Genesys Cloud integration facebook`,
+
+		CreateContext: provider.CreateWithPooledClient(createIntegrationFacebook),
+		ReadContext:   provider.ReadWithPooledClient(readIntegrationFacebook),
+		UpdateContext: provider.UpdateWithPooledClient(updateIntegrationFacebook),
+		DeleteContext: provider.DeleteWithPooledClient(deleteIntegrationFacebook),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 1,
+		Timeouts: &schema.ResourceTimeout{
+			Update: schema.DefaultTimeout(8 * time.Minute),
+			Read:   schema.DefaultTimeout(8 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			`name`: {
+				Description: `The name of the Facebook Integration`,
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+			`supported_content_id`: {
+				Description: `The SupportedContent unique identifier associated with this integration.`,
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			`messaging_setting_id`: {
+				Description: `The messaging Setting unique identifier associated with this integration.`,
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			`page_access_token`: {
+				Description: `The long-lived Page Access Token of Facebook page. 
+			See https://developers.facebook.com/docs/facebook-login/access-tokens. 
+			Either pageAccessToken or userAccessToken should be provided.`,
+				Optional: true,
+				Type:     schema.TypeString,
+			},
+			`user_access_token`: {
+				Description: `The short-lived User Access Token of the Facebook user logged into the Facebook app. 
+			See https://developers.facebook.com/docs/facebook-login/access-tokens. 
+			Either pageAccessToken or userAccessToken should be provided.`,
+				Optional: true,
+				Type:     schema.TypeString,
+			},
+			`page_id`: {
+				Description: `The page Id of Facebook page. The pageId is required when userAccessToken is provided.`,
+				Optional:    true,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+			},
+			`app_id`: {
+				Description: `The app Id of Facebook app. The appId is required when a customer wants to use their own approved Facebook app.`,
+				Optional:    true,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+			},
+			`app_secret`: {
```

Comment:
may be better to add a validation inside the createContext just to check all these conditions are met. (ideal place would be a validateFunc but there are challenges referencing other input varaible in a current variable context)

if appsecret has a value entered, appId should be present too. (viceversa)
pageaccesstoken and userAccessToken need not be simultaneously present 
pageid need not be provided if pageaccesstoken is provided and userAccessToken not provided